base/file.jl

```diff
@@ -315,16 +325,18 @@ end
 tempdir() = dirname(tempname())
 
 # Create and return the name of a temporary file along with an IOStream
-function mktemp(parent=tempdir())
-    b = joinpath(parent, "tmpXXXXXX")
+function mktemp(parent=tempdir(); prefix=temp_prefix)
+    format = prefix*"XXXXXX"
+    b = joinpath(parent, format)
     p = ccall(:mkstemp, Int32, (Cstring,), b) # modifies b
     systemerror(:mktemp, p == -1)
     return (b, fdio(p, true))
 end
 
 # Create and return the name of a temporary directory
-function mktempdir(parent=tempdir())
-    b = joinpath(parent, "tmpXXXXXX")
+function mktempdir(parent=tempdir(); prefix=temp_prefix)
+    format = prefix*"XXXXXX"
+    b = joinpath(parent, format)
     p = ccall(:mkdtemp, Cstring, (Cstring,), b)
```

Comment:
libuv now has a mkdtemp implementation, which could be used to reduce the number of code-paths we have to test for this function (http://docs.libuv.org/en/v1.x/fs.html#c.uv_fs_mkdtemp)